src/content/en/showcase/case-study/service-workers-iowa.markdown

```diff
@@ -0,0 +1,507 @@
+published_on: 2015-10-01
+updated_on: 2015-10-01
+authors:
+  - jeffposnick
+featured_image: images/service-workers-iowa/card.png
+showcase:
+  key_image: images/service-workers-iowa/screenshot-portrait.png
+  scores:
+    pagespeed:
+      speed: 65
+      ux: 92
+    webpagetest:
+      value: 4658
+      result: http://www.webpagetest.org/result/151001_R6_141S/
+  link: https://events.google.com/io2015/
+  overview:
+    description: |
+      This year’s [Google I/O 2015 web app](https://events.google.com/io2015/) was written by Google’s Developer Relations team, based on designs by our
+      friends at [Instrument](http://www.instrument.com/), who wrote the nifty [audio/visual experiment](http://www.instrument.com/news/google-io-2015). Our team’s mission was to ensure that the I/O web app (which I’ll refer to by
+      its codename, IOWA) showcased everything the modern web could do. A full
+      offline-first experience was at the top of our list of must-have features.
+
+      If you’ve read any of the other articles on this site recently, you’ve
+      undoubtedly encountered [service workers](http://www.html5rocks.com/en/tutorials/service-worker/introduction/), and you won’t be surprised to hear that IOWA’s offline support is heavily
+      reliant on them. Motivated by the real-world needs of IOWA, we developed two
+      libraries to handle two different offline use cases: [`sw-precache`](https://github.com/GoogleChrome/sw-precache) to automate precaching of static resources, and [`sw-toolbox`](https://github.com/GoogleChrome/sw-toolbox) to handle runtime caching and fallback strategies.
+
+      The libraries complement each other nicely, and allowed us to implement a
+      performant strategy in which IOWA’s static content “shell” was always served
+      directly from the cache, and dynamic or remote resources were served from the
+      network, with fallbacks to cached or static responses when needed.
+
+  sections:
+    - title: Precaching with <code>sw-precache</code>
+      content: |
+        IOWA’s static resources—its HTML, JavaScript, CSS, and images—provide the core
+        shell for the web application. There were two specific requirements that were
+        important when thinking about caching these resources: we wanted to make sure
+        that most static resources were cached, and that they were kept up to date.
+        [`sw-precache`](https://github.com/GoogleChrome/sw-precache) was built with those
+        requirements in mind.
+
+      subsections:
+        - title: Build-time Integration
+          content: |
+            <code>sw-precache</code> with IOWA’s [`gulp`](http://gulpjs.com/)-based build process, and we rely on a series of [glob](https://github.com/isaacs/node-glob) patterns to ensure that we generate a complete list of all the static
+            resources IOWA uses.
+
+            <figure>
+              <pre>staticFileGlobs: [
```

Comment:
I couldn't figure out how to get `{% highlight javascript %} {% include blah.js %} {% endhighlight %}` working when the content was coming from the YAML front matter, so I resorted to `<pre>` tags. I wish I could get syntax highlighting, though...